src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2195,6 +2195,44 @@ exprt c_typecheck_baset::do_special_functions(
 
     return bswap_expr;
   }
+  else if(identifier=="__builtin_fpclassify" ||
+          identifier==CPROVER_PREFIX "fpclassify")
+  {
+    if(expr.arguments().size()!=6)
```

Comment:
I'd add `const auto &arguments = expr.arguments();` as that function call is repeated a number of times in this function.